pyodata/v2/service.py

```diff
@@ -545,36 +552,45 @@ def set(self, **kwargs):
 class EntityDeleteRequest(ODataHttpRequest):
     """Used for deleting entity (DELETE operations on a single entity)"""
 
-    def __init__(self, url, connection, handler, entity_set, entity_key):
+    def __init__(self, url, connection, handler, entity_set, entity_key, encode_path=True):
         super(EntityDeleteRequest, self).__init__(url, connection, handler)
         self._logger = logging.getLogger(LOGGER_NAME)
         self._entity_set = entity_set
         self._entity_key = entity_key
+        self._encode_path = encode_path
 
         self._logger.debug('New instance of EntityDeleteRequest for entity type: %s', entity_set.entity_type.name)
 
     def get_path(self):
+        if self.get_encode_path():
+            return quote(self._entity_set.name + self._entity_key.to_key_string())
         return self._entity_set.name + self._entity_key.to_key_string()
 
+    def get_encode_path(self):
+        """Getter for encode path flag"""
+        return self._encode_path
+
     def get_method(self):
         # pylint: disable=no-self-use
         return 'DELETE'
 
 
+# pylint: disable=too-many-instance-attributes
 class EntityModifyRequest(ODataHttpRequest):
     """Used for modyfing entities (UPDATE/MERGE operations on a single entity)
 
        Call execute() to send the update-request to the OData service
        and get the modified entity."""
 
     ALLOWED_HTTP_METHODS = ['PATCH', 'PUT', 'MERGE']
-
-    def __init__(self, url, connection, handler, entity_set, entity_key, method="PATCH"):
+    # pylint: disable=too-many-arguments
+    def __init__(self, url, connection, handler, entity_set, entity_key, method="PATCH", encode_path=True):
```

Comment:
pyodata/v2/service.py:587:5: E301 expected 1 blank line, found 0
make: *** [Makefile:35: lint] Error 1

from linter build, 
run locally with `flake8 --config=.flake8  pyodata`

Reply:
To satisfy pylint and flake 8, it is enough to add new line, like this:

```
    ALLOWED_HTTP_METHODS = ['PATCH', 'PUT', 'MERGE']

    # pylint: disable=too-many-arguments
    def __init__(self, url, connection, handler, entity_set, entity_key, method="PATCH", encode_path=True)
```:

The PR was not allowed with commits from maintainers to the branch, @linda-sap pls update on your end. 

Otherwise the PR is looking good. If with no communication for a long time, I will merge it as is and fix in subsequent commit.